nl_linux: Set NETLINK_GET_STRICT_CHK on sockets

cjmakes opened the pull request:
Adds the GET_STRICT_CHK option on netlink sockets to avoid missing
routes due to a kernel bug[1]. The choice to set this option on all new
sockets copies the iproute2 behaviour fix[2]

[1]: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=885b8b4dbba5ca6114db0fcd0737fe2512650745
[2]: https://git.kernel.org/pub/scm/network/iproute2/iproute2.git/commit/ip?id=aea41afcfd6d6547dd2e80ddde8df7e3b2800482

vishvananda commented:
The iproute2 code explicitly ignores error returns since older kernels don't support this flag:

https://github.com/shemminger/iproute2/blob/7846496bf721293fd72836a3677a109530858d0e/lib/libnetlink.c#L169

I think we should probably just log the failure and continue instead of returning the error

cjmakes commented:
It makes sense to continue on. There is no logging elsewhere in library so it is probably best to just completely ignore the error.

cjmakes commented:
Setting the strict option for all operations exposed a couple bugs where the wrong message type was used when communicating with netlink. I fixed the bugs I found and changed this patch to add toggling the strict option. I think a method to set the option is more flexible to allow setting for some options now without needing to find an fix all message type issues at once.

Another bug I found as a result of this but have not had time to fix is #749

vishvananda commented:
thanks lgtm
